synapse/config/database.py

```diff
@@ -76,11 +135,24 @@ def read_arguments(self, args):
         self.set_databasepath(args.database_path)
 
     def set_databasepath(self, database_path):
```

Comment:
given that this is grandfathering in a deprecated config setting, shouldn't it work by operating on the dict before it is passed to `DatabaseConnectionConfig` rather than gut-wrenching things into the innards of `DatabaseConnectionConfig`?

Reply:
`set_databasepath` can be called after `read_config` due to the `--database-param` cli option, which makes it a bit of a PITA

Reply:
fair